src/riak_governor.erl

```diff
@@ -16,12 +16,19 @@ is_leader(Key) ->
     is_leader(node(), Key).
 
 is_leader(Node, Key) ->
+    MaybeLeader = get_leader(Node, Key),
+    MaybeLeader =/= no_leader andalso MaybeLeader =:= node().
+
+get_leader(Key) ->
+    get_leader(node(), Key).
+
+get_leader(Node, Key) ->
     DocIdx   = riak_core_util:chash_key(Key),
     Preflist = riak_governor_util:get_primary_apl(DocIdx),
     Nodes    = lists:usort([PrefNode || {{_Index, PrefNode}, _Type} <- Preflist]),
     case Nodes of
-        [Node] -> true;
-        [_] -> false;
+        [Node] -> Node;
+        [_] -> no_leader;
```

Comment:
This won't work.
`[_]` clause means that there is a different Node in group that the one passed to function.
But his function in general was ugly, so I already extracted get_leader: https://github.com/loucash/riak_governor/commit/968ce53dddce675f63a68dc3ccfe702008c5c419

(Sorry, I didn't expect this PR so quickly)